### PR TITLE
Added proper SteamID2 and IP address validation to admin ban page / Switch to userid for kickid

### PR DESCRIPTION
**Pull request by geominorai**
## Description
Replaces string length tests with regex tests
Fixes not allowing older Steam accounts with a shorter ID length
Includes alternate regex test for community IDs

Switched `kickid` to use `userid` instead of Steam ID due to command breakage for TF2 servers.

## Motivation and Context
- Prevents passing invalid Steam IDs and IP addresses that may cause errors later in the pipeline
- Fixes players not being kicked after being banned via the admin ban page or kicked via the server list.

## How Has This Been Tested?
Tested on Linux web server
Tested on Linux TF2 game server

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by geominorai** [comment]
Added commits to switch to `userid` for the calls to the `kickid` command.

**Comment by geominorai** [comment]
This will probably fix issue #474.

**Comment by geominorai** [comment]
Renamed the for-loop variables by @codacy-bot's recommendations.

**Comment by geominorai** [comment]
Fixed a bug where a SteamID2 conversion was attempted before checking for an empty string (such as for IP bans).

File: web/includes/sb-callback.php [hunk 1]
```diff
@@ -1672,8 +1672,9 @@ function AddBan($nickname, $type, $steam, $ip, $length, $dfile, $dname, $reason,
         Log::add("w", "Hacking Attempt", "$username tried to add a ban, but doesnt have access.");
         return $objResponse;
     }
-
-    $steam = \SteamID\SteamID::toSteam2(trim($steam));
+    $steam = trim($steam);
+    if ($steam)
```
Comment (CrazyHackGUT) [inline web/includes/sb-callback.php]: Use `empty()` function.

Comment (geominorai) [reply web/includes/sb-callback.php]: It's acts the same for strings, but alright.

**Comment by geominorai** [comment]
Pull has been abandoned and is too far behind master.  I'll resync and recreate the patch and pull request.

**Pull request closed without merging**